person.rb

```diff
@@ -22,4 +23,9 @@ def can_use_service?
   def correct_name
     @name
   end
+
+  def add_rental_details(date, book)
+    Rental.new(date, self, book)
+    rental_details.push(self)
+  end
```

Comment:
- [ ] The code on line 28 needs to match the Rental constructor parameter order. You should have this depending on the order of the parameters.

  ```
  def add_rental_details(date, book)
    Rental.new(date, book, self)
  end
  ```

- [ ] Kindly remove the code on line 29. Initializing a rental on line 28 already adds the rental to the `rental_details` array in the Rental class, so there is not need to add a duplicate.